core/blockchain.go

```diff
@@ -1404,7 +1377,7 @@ func (bc *BlockChain) InsertChainWithoutSealVerification(block *types.Block) (in
 // racey behaviour. If a sidechain import is in progress, and the historic state
 // is imported, but then new canon-head is added before the actual sidechain
 // completes, then the historic state could be pruned again
-func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, error) {
+func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals, catalystMode bool) (int, error) {
```

Comment:
Shouldn't this `catalystMode` be replaced with `merger.LeftPoW`? AFAIK that is the trigger for checking PoW or PoS rules. Even in catalyst mode, PoW part should be done with PoW rules?

Reply:
Or alternatively, shouldn't we internally look at the TTD and change the behavior based on which part of the chain we're processing? Unsure, asking really.